\documentclass[11pt, letterpaper]{article}   	% use "amsart" instead of "article" for AMSLaTeX format
\usepackage{graphicx}				% Use pdf, png, jpg, or epsÂ§ with pdflatex; use eps in DVI mode
\usepackage{amssymb}
\usepackage{amsmath}
\usepackage{url}
\usepackage[backend=biber, style=authoryear, autocite=inline, isbn=false, doi=false, url=false, eprint=false]{biblatex}
% Fix biblatex behavior that writes "In: " before journal name.
\renewbibmacro{in:}{%
  \ifentrytype{article}{}{\printtext{\bibstring{in}\intitlepunct}}}

\bibliography{references}

\title{Distinguishing among coalescent models using two-site allele frequency spectra}
\author{Daniel P. Rice}
\date{\today}

\begin{document}
\maketitle

\abstract{[NOTE: OLD]
The genetic diversity of a population reflects its demographic and
evolutionary history. Methods for inferring this history typically
assume that the ancestry of a sample can be modeled by the Kingman
coalescent process. A defining feature of the Kingman coalescent is
that it generates genealogies that are binary trees: no more than two
ancestral lineages may coalesce at the same time. However, this
assumption breaks down under several scenarios. For example, pervasive
natural selection, rapid spatial range expansion, and extreme
variation in offspring number can all generate genealogies with
``multiple-merger'' events in which more than two lineages coalesce
instantaneously. Therefore, detecting multiple mergers is important
both for understanding which forces have shaped the diversity of a
population and for avoiding fitting misspecified models to data.
Current methods to detect multiple mergers rely on the average site
frequency spectrum (SFS). However, the signatures of multiple
mergers in the average SFS are also consistent with a Kingman
coalescent process with a time-varying population size. Here, I
present a new method for detecting multiple mergers based on
the mutual information of the joint allele frequency spectrum at pairs of linked sites. Unlike the average SFS,
the mutual information depends mostly on the topologies of genealogies
rather than their branch lengths and is therefore robust to most
demographic effects.}

\section*{Introduction}
The genetic diversity of a population reflects its demographic and evolutionary history.
Learning about this history from contemporary genetic data is the domain of modern population genetics (see~\cite{Hahn2018a}).
The fundamental tools of the trade are toy models, which show how historical forces shape genetic diversity, and which form the basis of parametric inference methods.
However, populations are complicated and, moreover, varied in their complications.
No simple model can capture the processes governing every species' evolution, and fitting a misspecified model will generate misleading inferences.
It is therefore crucial to understand the limits of our models and to be able to assess when a model is appropriate for a particular data set.

One of the most important and widely used models is the Kingman coalescent~\autocite{Kingman1982a, Kingman1982b, Kingman1982c, Hudson1983, Tajima1983}.
The Kingman coalescent is a stochastic process that generates gene genealogies: trees representing the patterns of shared ancestry of sampled individuals.
Inference methods use these genealogies as latent variables linking demographic parameters to observable data~\autocite{RosenbergNordborg2002}.
The Kingman coalescent has a number of convenient properties that allow for both analytical calculations (e.g.,~\cite{Tajima1989}) and efficient stochastic simulations (e.g.,~\cite{ms})---tree topologies are independent of waiting times; waiting times are generated by a Markov process; and neutral mutations are modeled as a Poisson process conditionally independent of the tree.
Moreover, the model may be extended to study a variety of biological phenomena including recombination, population structure, and variation in sex ratios or ploidy (see generally~\cite{Wakeley2009}).

In its simplest form, the Kingman coalescent has a single parameter, the coalescent rate, which determines the branch lengths of genealogies~\autocite{Kingman1982a}.
Under a variety of conditions, the coalescent rate is inversely proportional to the size of the population~\autocite{Kingman1982b}.
Accordingly, a growing or shrinking population may be modeled by a time-varying rate~\autocite{GriffithsTavare1994, GriffithsTavare1998}.
This observation is the basis for a variety of methods to infer the population size history from genetic data.

With modern population genomics data sets, full-data likelihood models are impractical, so population-size inference is typically done on informative summary statistics.
One widely used statistic is the site frequency spectrum (SFS): the number of mutations observed as a function of their allele frequency in the sample.
The expected number of mutations at a given frequency depends on the branch lengths, and hence, on the coalescent rate.
Thus, one can infer the coalescent rate by integrating over tree topologies, weighted by their probabilities under the Kingman coalescent.
Some methods perform this integration by Monte Carlo simulation (e.g., \cite{CoventryEtAl2010, ExcoffierEtAl2013}).
Others (e.g., \cite{Nielsen2000}, \cite{BhaskarEtAl2015}), compute the expected site frequency spectrum directly for simple demographic models using the results of \cite{GriffithsTavare1998} or \cite{PolanskiKimmel2003}.
(Another class of SFS-based methods are based on forward-time models rather than the coalescent \autocite{GutenkunstEtAl2009, LukicEtAl2011, RagsdaleGutenkunst2017, JouganousEtAl2017}.)

[Question: where do I cite the SMC models?]

Model misspecification. Schrider, others on problems of inference. Evidence for widespread selection.

Class of non-Kingman coalescents: multiple mergers. Where they arise. What they mean for inference (e.g. distort SFS, long-range linkage, timescale issues)

Tests for multiple mergers and non-neutrality in general.
General neutrality tests (Tradition of Taj's D. Achaz)
Problem with confounding (compare to neutral background)
Local tests for sweeps (haplotype tests)
Specific tests for multiple mergers

Our test, based on the 2-site frequency spectrum:
\begin{itemize}
\item Genome-wide. Not comparing loci to putatively neutral background.
\item Distinguishes time-varying N from multiple mergers.
\item SNP data.
\item Doesn't require phasing or ancestral states.
\item Easy to compute.
\item Works in principle with small samples.
\end{itemize}

Outline:
\begin{itemize}
\item Define statistic, show that it works without recombination. (Detects multiple mergers, doesn't respond to population size changes)
\item Coalescent simulations with recombination (it still works).
\item Forward-time simulations of linked sweeps (still works).
\item Application to D. mel (it detects non-Kingman)
\end{itemize}

Modern population genetics makes it possible to learn the evolutionary and demographic history of a population from its contemporary genetic diversity~\autocite{Hahn2018}.
The genetic differences among individuals reflect their genealogical history, which, in turn, reflects the history of various population parameters such as the population size, spatial structure, and the influence of natural selection.
By modeling how the (unobserved) genealogical process depends on these parameters and how patterns of genetic diversity depend on genealogy, we can develop inference procedures to learn about the past.

The most commonly used genealogical model is the Kingman coalescent, which arises in many models of neutral evolution~\autocite{Kingman1982, Hudson1983, Tajima1983}.
The key characteristic of the Kingman coalescent is that only two lineages are permitted to share a common ancestor at the same point in time.
Moreover, the genealogies generated by the Kingman coalescent are randomly bifurcating trees: each node subtends exactly two branches and the number of leaves subtended by one of these branches is uniformly distributed~\autocite{Tajima1983}.

% Kingman coalescent models are appropriate when a single timescale controls patterns of relatedness between individuals in a sample from a population.
% This timescale, which we will call $T_2$, is inversely proportional to the rate of coalescence between pairs of lineages and sets the branch lengths of genealogies.
% Thus, $T_2$ determines observable quantities such as the average genetic diversity between pairs of individuals and length distribution of genomic segments that are identical-by-descent.
% Population genetic inference methods interpret these quantities in light of the Kingman distribution of tree topologies to estimate the coalescent timescale.
% In the simplest models of neutral evolution, the coalescent timescale is proportional to the number of individuals in the population. Therefore, it is commonly referred to as the effective population size.

%The Kingman coalescent generates genealogies with several distinguishing characteristics. First, the genealogies are binary trees: each node has only two children. A second and related topological feature is that for a node subtending $n$ leaves, the number of leaves subtended by one of its children is uniformly distributed between 1 and $n-1$. Finally, under a neutral model of evolution, the rate of coalescence (and thus the branch lengths of the genealogy) is inversely proportional to the population size.

Because computing the full likelihood of the data is generally intractable, population genetic inference is typically done on informative summary statistics.
One informative statistic is the site frequency spectrum (SFS): the number of mutations observed as a function of their allele frequency in a sample.
The expected number of mutations in $i$ sampled chromosomes is proportional to the lengths of branches subtending $i$ leaves of the genealogy, and thus depends the distributions of topologies as well as branch lengths.
By assuming the Kingman model, one can marginalize over the unobserved tree topologies and extract information about the distribution of branch lengths and thus about the rate of coalescence as a function of time.
If this assumption is good, the site frequency spectrum thus reflects the history of population size changes.
For example, high-frequency alleles are the result of mutations on deep branches and their number in a sample reflects the population size in the distant past.

A serious problem for this inference procedure is that alternative models of evolution generate different coalescent models.
For example, models of pervasive weak selection, skewed offspring number distributions, and periodic strong bottlenecks can all generate genealogies that differ from the Kingman both topologically and in terms of branch length.
[Review of MMC literature here? Neutral models (Eldon, Birkner, etc.), sweeps (Durett Schweinsberg, Coop and Ralph), rapid adaptation (Neher, Desai), background selection (Seger, Good, Nicholaisen), Range expansion (??)]
Collectively, these models are known as \emph{multiple mergers coalescents} because, unlike the Kingman coalescent, they permit multiple lineages to coalesce instantaneously.
As a result, multiple-mergers genealogies contain nodes with more than two children and lack the uniform branching structure of the Kingman.
Even more important for inference is the fact that the coalescent timescales---and thus the branch lengths and levels of diversity---are determined by parameters other than the population size.
In this case, the term `effective population size' is misleading because $T_2$ is no longer proportional to the number of individuals in the population.
Thus, the results of a Kingman-based inference procedure applied to data generated by a non-Kingman process will be both quantitatively and qualitatively wrong.
Clearly, it would be useful to be able to distinguish among different coalescent models, both to check that popular inferences of historical population sizes are valid and to understand which evolutionary forces are dominant in various populations.

[Mention the distinction between single-locus models (e.g. Seger on Whale lice, Neher et al. with viruses) where you can explicitly infer trees, and multi-locus methods, where you average over many latent genealogies. This method is in the latter group.]

An obvious question is whether we can use the average site frequency spectrum to reject multiple-mergers.
There are two distinguishing features of the SFS generated by multiple mergers coalescents.
First, low-frequency mutations are enriched relative to the time-homogeneous Kingman coalescent.
Unfortunately, this is also a feature of the Kingman coalescent in growing populations, so it is not well-suited for model selection (but see \cite{}).
Second, the multiple-merger SFS is non-monotonic: it has a positive slope for high frequencies.
Unlike the excess of rare alleles, this feature cannot be reproduced by an exchangeable Kingman coalescent model.
However, it is difficult to identify the non-monotonicity in real data.
The problem is that it necessary to know which allele is ancestral to distinguish derived mutations at frequency $x$ from those at frequency $1-x$.
Various methods for identifying the ancestral allele exist, but in practice even a very low rate of error can generate a spurious signal.

Ideally, we would like a statistic (or statistics) that:
\begin{enumerate}
\item is qualitatively different between models of genealogies,
\item is robust to demographic changes, and
\item requires minimal information beyond genetic diversity.
\end{enumerate}
The first two criteria are important for making qualitative distinctions between models. The last one would maximize its applicability.
Here, I propose that the joint site frequency spectrum at pairs of linked sites is such a statistic.

\section*{Single-locus theory}

\section*{Recombination}

\section*{Application to \textit{Drosophila melanogaster}}

\printbibliography

\end{document}

TODO:
- Review and re-write abstract.
-
