\documentclass[11pt, letterpaper]{article}   	% use "amsart" instead of "article" for AMSLaTeX format
\usepackage{graphicx}				% Use pdf, png, jpg, or epsÂ§ with pdflatex; use eps in DVI mode
\usepackage{amssymb}
\usepackage{amsmath}
\usepackage{url}
\usepackage[backend=biber, style=authoryear, autocite=inline, isbn=false, doi=false, url=false, eprint=false]{biblatex}
% Fix biblatex behavior that writes "In: " before journal name.
\renewbibmacro{in:}{%
  \ifentrytype{article}{}{\printtext{\bibstring{in}\intitlepunct}}}

\bibliography{references}

\title{Distinguishing among coalescent models using two-site allele frequency spectra}
\author{Daniel P. Rice}
\date{\today}

\begin{document}
\maketitle

\abstract{[NOTE: OLD]
The genetic diversity of a population reflects its demographic and
evolutionary history. Methods for inferring this history typically
assume that the ancestry of a sample can be modeled by the Kingman
coalescent process. A defining feature of the Kingman coalescent is
that it generates genealogies that are binary trees: no more than two
ancestral lineages may coalesce at the same time. However, this
assumption breaks down under several scenarios. For example, pervasive
natural selection, rapid spatial range expansion, and extreme
variation in offspring number can all generate genealogies with
``multiple-merger'' events in which more than two lineages coalesce
instantaneously. Therefore, detecting multiple mergers is important
both for understanding which forces have shaped the diversity of a
population and for avoiding fitting misspecified models to data.
Current methods to detect multiple mergers rely on the average site
frequency spectrum (SFS). However, the signatures of multiple
mergers in the average SFS are also consistent with a Kingman
coalescent process with a time-varying population size. Here, I
present a new method for detecting multiple mergers based on
the mutual information of the joint allele frequency spectrum at pairs of linked sites. Unlike the average SFS,
the mutual information depends mostly on the topologies of genealogies
rather than their branch lengths and is therefore robust to most
demographic effects.}

\section*{Introduction}
The genetic diversity of a population reflects its demographic and evolutionary history.
Learning about this history from contemporary genetic data is the domain of modern population genetics (see~\cite{Hahn2018a}).
The fundamental tools of the trade are toy models, which show how historical forces shape genetic diversity, and which form the basis of parametric inference methods.
However, populations are complicated and, moreover, varied in their complications.
No simple model can capture the processes governing every species' evolution, and fitting a misspecified model will generate misleading inferences.
It is therefore crucial to understand the limits of our models and to be able to assess when a model is appropriate for a particular data set.

One of the most important and widely used models is the Kingman coalescent~\autocite{Kingman1982a, Kingman1982b, Kingman1982c, Hudson1983, Tajima1983}.
The Kingman coalescent is a stochastic process that generates gene genealogies: trees representing the patterns of shared ancestry of sampled individuals.
Inference methods use these genealogies as latent variables linking demographic parameters to observable data~\autocite{RosenbergNordborg2002}.
The Kingman coalescent has a number of convenient properties that allow for both analytical calculations (e.g.,~\cite{Tajima1989}) and efficient stochastic simulations (e.g.,~\cite{Hudson2002})---tree topologies are independent of waiting times; waiting times are generated by a Markov process; and neutral mutations are modeled as a Poisson process conditionally independent of the tree.
Moreover, the model may be extended to study a variety of biological phenomena including recombination, population structure, and variation in sex ratios or ploidy (see generally~\cite{Wakeley2009}).

In its simplest form, the Kingman coalescent has a single parameter, the coalescent rate, which determines the branch lengths of genealogies~\autocite{Kingman1982a}.
Under a variety of conditions, the coalescent rate is inversely proportional to the size of the population~\autocite{Kingman1982b}.
Accordingly, a growing or shrinking population may be modeled by a time-varying rate~\autocite{GriffithsTavare1994, GriffithsTavare1998}.
This observation is the basis for a variety of methods to infer historical population sizes from genetic data.

With modern population genomics data sets, full-data likelihood models are impractical, so population-size inference is typically done on informative summary statistics.
One widely used statistic is the site frequency spectrum (SFS): the number of mutations observed as a function of their allele frequency in the sample.
The expected number of mutations at a given frequency depends on the branch lengths, and hence, on the coalescent rate.
Thus, one can infer the coalescent rate by integrating over tree topologies, weighted by their probabilities under the Kingman coalescent.
Some methods perform this integration by Monte Carlo simulation (e.g., \cite{CoventryEtAl2010, ExcoffierEtAl2013}).
Others (e.g., \cite{Nielsen2000}, \cite{BhaskarEtAl2015}), compute the expected site frequency spectrum directly for simple demographic models using the results of \cite{GriffithsTavare1998} or \cite{PolanskiKimmel2003}.
(Another class of SFS-based methods are based on corresponding forward-time models rather than the coalescent \autocite{GutenkunstEtAl2009, LukicEtAl2011, RagsdaleGutenkunst2017, JouganousEtAl2017}.)

A serious problem for this inference procedure is that different models of evolution generate different relationships between historical population sizes and genetic diversity.
One of the basic assumptions of the Kingman coalescent is that natural selection is negligible in determining the distribution of genealogies.
When this assumption is violated, Kingman-based inference methods are misspecified.
For example, \cite{SchriderEtAl2016} recently demonstrated how population size inference can be distorted by selective sweeps.
This effect is present in SFS-based methods as well as in sequentially Markov coalescent methods (e.g., \cite{LiDurbin2011}).
In a similar vein, \cite{CvijovicEtAl2018}, showed that purifying selection at linked sites that is sufficient to reduce overall genetic diversity is also sufficient to distort the SFS, leading to a false signal of population growth.
Genomic evidence from multiple species suggests that such violations of the neutral model underlying the Kingman coalescent may be widespread \autocite{SellaEtAl2009, Corbett-DetigEtAl2015, KernHahn2018}.

An important extension of the Kingman coalescent is a family of models known as \textit{multiple merger coalescents} \autocite{Pitman1999, Sagitov1999, DonnellyKurtz1999,} (reviewed in \cite{Eldon2016}), which arise in a variety of neutral and non-neutral contexts.
Whereas in the Kingman coalescent lineages may only coalesce pairwise, multiple merger coalescents permit more than two lineages to coalesce in a single event.
The even more general class of simultaneous multiple merger coalescents \autocite{Schweinsberg2000, MohleSagitov2001, Sagitov2003} permits more than one multiple merger event at the same time.
These models are relevant for species with
``sweepstakes'' reproductive events \autocite{EldonWakeley2006, SargsyanWakeley2008},
fat-tailed offspring number distributions \autocite{Schweinsberg2003},
recurring selective sweeps at linked sites \autocite{DurrettSchweinsberg2005, CoopRalph},
rapid adaptation \autocite{NeherHallatscheck2013, DesaiEtAl},
and purifying selection at sufficiently dense sites \autocite{Seger, Good, Nicholaisen}.
In each of these contexts, the coalescent timescale is not necessarily proportional to the population size.
For example, with fat-tailed offspring distributions the rate of coalescence is a power law in the population size \autocite{Schweinsberg2003}, while with linked sweeps it is determined by rate of linked sweeps \autocite{DurrettSchweinsberg2005}.
In these settings, interpreting the level of genetic diversity in terms of an ``effective population size'' is misleading and inferences based on the Kingman coalescent may be \emph{qualitatively} incorrect.
It is therefore important to determine whether the Kingman model is appropriate for a given data set before performing demographic inference.

One possible approach to identifying multiple mergers in genomic data to use the SFS as a summary statistic.
(\cite{Koskela2015} developed a full-likelihood method based on importance sampling, but as with other ``exact'' inference methods it does not scale to genomic data.)
\cite{BirknerEtAl2013, BlathEtAl2016, SpenceEtAl2016} derived methods for computing the expected site frequency spectrum of (simultaneous-)multiple merger coalescents.
\cite{EldonEtAl2015} showed that it is possible to distinguish between a multiple merge coalescent of the beta family and the Kingman coalescent with exponential growth using the SFS.
\cite{RodelspergerEtAl2014} detected non-Kingman signatures of widespread linked selection in the nematode \textit{Pristionchus pacificus} by demonstrating that the site frequency spectrum is non-monotonic, a signature of multiple mergers \autocite{NeherHallatscheck2013, BirknerEtAl2013}.

However, existing methods have limitations for distinguishing multiple mergers from general models of population-size change.
The primary signature of multiple mergers in the SFS is an overabundance of low-frequency mutations relative to the Kingman expectation, which is also the signature of population growth.
\cite{EldonEtAl2015} were able to reject exponential growth in favor of multiple mergers with sufficient data, but a more flexible model of growth may be able to fit the multiple mergers SFS (see \cite{MyersEtAl2008, BhaskarSong2014}).
The non-mononotic SFS identified by \cite{RodelspergerEtAl2014} is a more robust signature of multiple mergers, but detecting it in data requires knowledge of the ancestral allele at each site.
High-frequency mutations are typically much rarer than low-frequency mutations, so misidentifying even a small fraction of ancestral alleles can generate a non-monotonic SFS at high frequencies.

Here, we propose that summary statistics based on the spectrum of mutation frequencies at pairs of nearby site---the 2-SFS \autocite{Hudson2001, FerrettiEtal2018})---are useful for distinguishing between the Kingman coalescent with population growth and multiple merger coalescents.
We show that this is true for both perfectly linked sites and sites separated by a moderate genetic distance.
These statistics may be calculated efficiently from genomic single nucleotide polymorphism data.
Furthermore, they do not require phasing, recombination maps, or ancestral allele identification and are informative even with small sample sizes.
Together, these properties make the 2-SFS useful for demographic model-checking in a wide range of species.
Finally, we demonstrate this model-checking procedure on genomic diversity data from \textit{Drosophila melanogaster} \autocite{LackEtAl2015}.

\section*{Definitions and single-locus theory}
[Define statistic, show that it works without recombination. (Detects multiple mergers, doesn't respond to population size changes)]

\section*{Simulations with recombination}
[Coalescent simulations with recombination (it still works).]

\section*{Linked selective sweeps}
[Forward-time simulations of linked sweeps (still works).]

\section*{Application to \textit{Drosophila melanogaster}}

\section*{Discussion}
\begin{itemize}
    \item Focus here on coalescent, but also relevant for diffusion methods
    \item Single-locus tests that rely on trees (Seger whale lice, Neher with viruses)
    \item Multi-locus tests that compare vs genomic background
    \item Could extend to a local method to look for variation in ``multiple-merger-ness''
    \item Could also extend
\end{itemize}

Tests of the standard Kingman coalescent based on the SFS originated with \cite{Tajima1989}, \cite{FuLi1993}, and \cite{SimonsenEtAl1995} (see \cite{Achaz2009} for a unifying framework).

\printbibliography

\end{document}

TODO:
- Review and re-write abstract.
-
