\documentclass[11pt, letterpaper]{article}   	% use "amsart" instead of "article" for AMSLaTeX format
%\usepackage{geometry}                		% See geometry.pdf to learn the layout options. There are lots.
%\geometry{letterpaper}                   		% ... or a4paper or a5paper or ... 
%\geometry{landscape}                		% Activate for rotated page geometry
%\usepackage[parfill]{parskip}    		% Activate to begin paragraphs with an empty line rather than an indent
\usepackage{graphicx}				% Use pdf, png, jpg, or epsÂ§ with pdflatex; use eps in DVI mode
								% TeX will automatically convert eps --> pdf in pdflatex		
\usepackage{amssymb}
\usepackage{amsmath}

%SetFonts

\newcommand{\eq}[1]{Eq.~(\ref{#1})}
\newcommand{\Eq}[1]{Equation~(\ref{#1})}
\newcommand{\fig}[1]{Fig.~\ref{#1}}
\newcommand{\Fig}[1]{Figure~\ref{#1}}
\newcommand{\E}[1]{\left<{#1}\right>}

\title{Distinguishing among coalescent models using two-site joint allele frequency spectra}
\author{Daniel P. Rice}
\date{\today}							

\begin{document}
\maketitle

\abstract{The genetic diversity of a population reflects its demographic and
evolutionary history. Methods for inferring this history typically
assume that the ancestry of a sample can be modeled by the Kingman
coalescent process. A defining feature of the Kingman coalescent is
that it generates genealogies that are binary trees: no more than two
ancestral lineages may coalesce at the same time. However, this
assumption breaks down under several scenarios. For example, pervasive
natural selection, rapid spatial range expansion, and extreme
variation in offspring number can all generate genealogies with
"multiple-merger" events in which more than two lineages coalesce
instantaneously. Therefore, detecting multiple mergers is important
both for understanding which forces have shaped the diversity of a
population and for avoiding fitting misspecified models to data.
Current methods to detect multiple mergers rely on the average site
frequency spectrum (SFS). However, the signatures of multiple
mergers in the average SFS are also consistent with a Kingman
coalescent process with a time-varying population size. Here, I
present a new method for detecting multiple mergers based on
the mutual information of the joint site frequency spectrum at pairs of linked sites. Unlike the average SFS,
the mutual information depends mostly on the topologies of genealogies
rather than their branch lengths and is therefore robust to most
demographic effects.}

\section*{Background}
A typical goal of population genetics is to infer the evolutionary and demographic history of a population from its contemporary genetic diversity. The genetic differences among individuals reflect their genealogical history, which, in turn, reflects the history of various population parameters such as the population size, spatial structure, and the influence of natural selection. By modeling how the (unobserved) genealogical process depends on these parameters and how patterns of genetic diversity depend on genealogy, we can develop inference procedures to learn about the past.

The most commonly used genealogical model is the Kingman coalescent, which arises in many models of neutral evolution.
The key characteristic of the Kingman coalescent is that only two lineages are permitted to share a common ancestor at the same point in (continuous) time.
As a result, the genealogies generated by the Kingman coalescent are randomly bifurcating trees: each node subtends exactly two branches and the number of leaves subtended by one of these branches is uniformly distributed.

Kingman coalescent models are appropriate when a single timescale controls patterns of relatedness between individuals in a sample from a population.
This timescale, which we will call $T_2$, is inversely proportional to the rate of coalescence between pairs of lineages and sets the branch lengths of genealogies.
Thus, $T_2$ determines observable quantities such as the average genetic diversity between pairs of individuals and length distribution of genomic segments that are identical-by-descent.
Population genetic inference methods interpret these quantities in light of the Kingman distribution of tree topologies to estimate the coalescent timescale.
In the simplest models of neutral evolution, the coalescent timescale is proportional to the number of individuals in the population. Therefore, it is commonly referred to as the effective population size.

%The Kingman coalescent generates genealogies with several distinguishing characteristics. First, the genealogies are binary trees: each node has only two children. A second and related topological feature is that for a node subtending $n$ leaves, the number of leaves subtended by one of its children is uniformly distributed between 1 and $n-1$. Finally, under a neutral model of evolution, the rate of coalescence (and thus the branch lengths of the genealogy) is inversely proportional to the population size.

Because computing the full likelihood of the data is generally intractable, population genetic inference is typically done on informative summary statistics.
One informative statistic is the site frequency spectrum (SFS): the number of mutations observed as a function of their allele frequency in a sample.
The expected number of mutations in $i$ sampled chromosomes is proportional to the lengths of branches subtending $i$ leaves of the genealogy, and thus depends the distributions of topologies as well as branch lengths.
By assuming the Kingman model, one can marginalize over the unobserved tree topologies and extract information about the distribution of branch lengths and thus about the rate of coalescence as a function of time.
If this assumption is good, the site frequency spectrum thus reflects the history of population size changes.
For example, high-frequency alleles are the result of mutations on deep branches and their number in a sample reflects the population size in the distant past. 

A serious problem for this inference procedure is that alternative models of evolution generate different coalescent models.
For example, models of pervasive weak selection, skewed offspring number distributions, and periodic strong bottlenecks can all generate genealogies that differ from the Kingman both topologically and in terms of branch length.
Collectively, these models are known as \emph{multiple mergers coalescents} because, unlike the Kingman coalescent, they permit multiple lineages to coalesce instantaneously.
As a result, multiple-mergers genealogies contain nodes with more than two children and lack the uniform branching structure of the Kingman.
Even more important for inference is the fact that the coalescent timescales---and thus the branch lengths and levels of diversity---are determined by parameters other than the population size.
In this case, the term `effective population size' is misleading because $T_2$ is no longer proportional to the number of individuals in the population.
Thus, the results of a Kingman-based inference procedure applied to data generated by a non-Kingman process will be both quantitatively and qualitatively wrong.
Clearly, it would be useful to be able to distinguish among different coalescent models, both to check that popular inferences of historical population sizes are valid and to understand which evolutionary forces are dominant in various populations.

An obvious question is whether we can use the average site frequency spectrum to reject the Kingman model?
There are two distinguishing features of the SFS generated by multiple mergers coalescents.
First, low-frequency mutations are enriched relative to the time-homogeneous Kingman coalescent.
Unfortunately, this is also a feature of the Kingman coalescent in growing populations, so it is not well-suited for model selection (but see \cite{}).
Second, the multiple-merger SFS is non-monotonic: it has a positive slope for high frequencies.
Unlike the excess of rare alleles, this feature cannot be reproduced by an exchangeable Kingman coalescent model.
However, it is difficult to identify the non-monotonicity in real data.
The problem is that it necessary to know which allele is ancestral to distinguish derived mutations at frequency $x$ from those at frequency $1-x$.
Various methods for identifying the ancestral allele exist, but in practice even a very low rate of error can generate a spurious signal.

Ideally, we would like a statistic (or statistics) that:
\begin{enumerate}
\item is qualitatively different between models of genealogies,
\item is robust to demographic changes, and 
\item requires minimal information beyond genetic diversity.
\end{enumerate}
The first two criteria are important for making qualitative distinctions between models. The last one would maximize its applicability.
Here, I propose that the joint site frequency spectrum at pairs of linked sites is such a statistic.

\section*{Theoretical motivation}

Consider a sample of $n$ chromosomes.
Denote the site frequency spectrum, the fraction of sites with a mutation in $i$ of $n$ samples, by $\phi_i$.
We define the joint site frequency spectrum, $\phi_{ij}(d)$ for pairs of sites separated by a genomic distance of $d$ bases to be the fraction of such pairs with a mutation in $i$ samples at the left site and a mutation in $j$ samples at the right site.
As described above, the expected site frequency spectrum is related to the expected branch lengths of the genealogies by:
\begin{equation}
\E{\phi_i} = \mu \E{\tau_i},
\end{equation}
where $\mu$ is the per--base pair mutation rate, $\tau_i$, is the total branch length subtending $i$ leaves, and expectations are taken over the distribution of genealogies.
Similarly, the joint site frequency spectrum is proportional to the second moments of the joint branch length distribution:
\begin{equation}
\E{\phi_{ij}} = \mu^2 \E{\tau_i^l \tau_j^r},
\end{equation}
where $\tau_i^l$ and $\tau_j^r$ are the branch lengths at the left and right site respectively.
There are several previous results that suggest that the correlation structure of branch lengths contains information that can distinguish multiple mergers from Kingman coalescent processes.
If this is the case, then the joint site frequency spectrum will be a useful summary statistic for coalscent model selection.

\subsubsection*{Tightly-linked sites}

For tightly-linked sites (i.e. $rdT_2\ll1$ for per--base pair recombination rate $r$), $\E{\tau_i^l \tau_j^r} \approx \E{\tau_i \tau_j}$, the branch length correlation matrix for a non-recombining locus.
This correlation matrix depends on correlations among the timing of coalescent events as well as correlations between clade sizes in tree topologies.
Fu \cite{} calculated the first and second moments of $\tau_i$ for the time-homogeneous Kingman coalescent.
His results show that $\E{\tau_i \tau_j} - \E{\tau_i}\E{\tau_j} < 0$ for $i \neq j$ and $i \neq n-j$.
This implies that the off-diagonal elements of the joint site frequency spectrum will be depleted relative to the product of the marginal site frequency spectra for tightly-linked sites.
Recently, Birkner \textit{et al.} \cite{}, extended Fu's calculation to the class of multiple-mergers coalescent models known as time-homogeneous $\Lambda$-coalescents.
In contrast to the Kingman case, Birkner \textit{et al.} found that, in the $\Lambda$-coalescent, $\E{\tau_i \tau_j} - \E{\tau_i}\E{\tau_j}$ may be positive for $i \neq j$.
These positive associations are largest for $1 < i - j \ll n$.
Together these results suggest that the joint site frequency spectrum of tightly-linked sites can distinguish qualitatively between the constant-sized Kingman and multiple-mergers coalescent.

We can get an intuitive understanding for this result by considering a sample of size 4. In the Kingman coalescent, there are only two possible tree topologies (\Fig{1}).
Furthermore, the total branch length is independent of the topology.
As a result, there is a trade-off between the branch length leading to singleton/tripleton mutations and the branch length leading to doubletons: loci with topology (A) will have more opportunities for the former and loci with topology (B) will have more opportunities for the latter.
Conditional on observing a doubleton at a locus, it is thus more likely that the locus has topology (B) and so the expected number of singletons is lower.
In terms of the joint site frequency spectrum, we have $\E{\phi_{12}} < \E{\phi_{1}} \E{\phi_{2}}$.

On the other hand, multiple mergers induce correlations between the tree topology and the total branch length.
For example, topology (C) has less overall opportunity for singletons as well as doubletons than (A) or (B), even though the expected proportion of singletons is higher.
Thus, observing \emph{any mutation at all} makes topology (C) less likely and the expected number of other mutations at all frequencies more likely.
If multiple-mergers events are frequent enough, this effect may dominate the tradeoff between (A) and (B) and lead to $\E{\phi_{12}} > \E{\phi_{1}} \E{\phi_{2}}$.

There are two limitations of these existing results.
First, they only apply to time-homogeneous coalescent processes and thus do not distinguish between multiple-mergers coalescents and Kingman models with time-varying population size.
Second, they assume a non-recombining locus and may break down for sites separated by a finite distance.

\subsubsection*{Loosely-linked sites}

Previous work on the multiple-mergers coalescent with recombination is limited relative to the literature on multiple-mergers models in non-recombining loci.
Nonetheless, existing results suggest that the two-site joint site frequency spectrum will behave differently as a function of genetic distance in multiple-mergers versus Kingman models.

Eldon and Wakeley \cite{} calculated the expected linkage disequilibrium in a coalescent model of a population with multiple-mergers induced by occasional ``jackpot'' reproductive events.
The found that LD may be reduced by multiple mergers at short genetic distances because of the elevated proportion of singleton mutations.
This effect is similar to the impact of population growth on LD.
In contrast, Eldon and Wakeley found that LD

LD in the MMC.

Correlations in coalescence times.

\section*{Summary statistics}

\section*{The jSFS-PMI distinguishes multiple-mergers from Kingman coalescent}

\section*{The signal is robust to population size change}

\section*{Low-recombination rate regions show signals of multiple mergers in \textit{D. melanogaster}}

\section*{Humans}

\end{document}

