{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Gazave et al. 2014 neutral region data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "D. P. Rice\n",
    "\n",
    "3/14/18 - 3/15/18"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Introduction\n",
    "\n",
    "This is an initial exploration of the data from Gazave et al. 2014, which sequenced ~500 European individuals to high coverage at 15 putatively neutral loci. The data consist of:\n",
    "1. __The target region definitions__ I've extracted these using `tesseract` OCR from Table S2\n",
    "2. __The SNVs__ I've downloaded a vcf from Alon Kleinan's website and extracted the allele counts relevant fields into a simple text file using `bcftools`\n",
    "\n",
    "The goal of this document is to perform the following quality checks:\n",
    "1. The numbers in the target regions should match those in table S2 (checking for OCR errors).\n",
    "2. Regions should make sense: target length = end - start, covered bases <= length.\n",
    "3. Region descriptive statistics should match main text and SI claims.\n",
    "4. SNV positions should be within target regions.\n",
    "5. Descriptive statistics of SNVs should match claims in the paper.\n",
    "6. Site frequency spectrum should match Fig. 2."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Target regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>length</th>\n",
       "      <th>n20x_1_indiv</th>\n",
       "      <th>n20x_450_indiv</th>\n",
       "      <th>nSNVs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>237146097</td>\n",
       "      <td>237165997</td>\n",
       "      <td>19900</td>\n",
       "      <td>19902</td>\n",
       "      <td>19764</td>\n",
       "      <td>180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>237360801</td>\n",
       "      <td>237380801</td>\n",
       "      <td>20000</td>\n",
       "      <td>20004</td>\n",
       "      <td>19539</td>\n",
       "      <td>197</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>4</td>\n",
       "      <td>164000000</td>\n",
       "      <td>164020000</td>\n",
       "      <td>20000</td>\n",
       "      <td>17962</td>\n",
       "      <td>16117</td>\n",
       "      <td>96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>164115673</td>\n",
       "      <td>164135673</td>\n",
       "      <td>20000</td>\n",
       "      <td>19768</td>\n",
       "      <td>19443</td>\n",
       "      <td>119</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6</td>\n",
       "      <td>165342001</td>\n",
       "      <td>165355001</td>\n",
       "      <td>13000</td>\n",
       "      <td>11098</td>\n",
       "      <td>10659</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   chr      start        end  length  n20x_1_indiv  n20x_450_indiv  nSNVs\n",
       "0    1  237146097  237165997   19900         19902           19764    180\n",
       "1    1  237360801  237380801   20000         20004           19539    197\n",
       "2    4  164000000  164020000   20000         17962           16117     96\n",
       "3    4  164115673  164135673   20000         19768           19443    119\n",
       "4    6  165342001  165355001   13000         11098           10659     86"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "targets = pd.read_table('../data/GazaveEtal/gazave_targets.txt', delim_whitespace=True, thousands=',')\n",
    "targets.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>length</th>\n",
       "      <th>n20x_1_indiv</th>\n",
       "      <th>n20x_450_indiv</th>\n",
       "      <th>nSNVs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>15.000000</td>\n",
       "      <td>1.500000e+01</td>\n",
       "      <td>1.500000e+01</td>\n",
       "      <td>15.000000</td>\n",
       "      <td>15.000000</td>\n",
       "      <td>15.000000</td>\n",
       "      <td>15.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>6.800000</td>\n",
       "      <td>1.364851e+08</td>\n",
       "      <td>1.364995e+08</td>\n",
       "      <td>14416.000000</td>\n",
       "      <td>13667.666667</td>\n",
       "      <td>13065.866667</td>\n",
       "      <td>122.266667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>3.233751</td>\n",
       "      <td>5.854636e+07</td>\n",
       "      <td>5.854773e+07</td>\n",
       "      <td>5279.405811</td>\n",
       "      <td>4936.985411</td>\n",
       "      <td>4731.881019</td>\n",
       "      <td>35.531609</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>1.000000</td>\n",
       "      <td>4.956573e+07</td>\n",
       "      <td>4.957973e+07</td>\n",
       "      <td>5340.000000</td>\n",
       "      <td>5285.000000</td>\n",
       "      <td>5066.000000</td>\n",
       "      <td>79.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>5.000000</td>\n",
       "      <td>8.329806e+07</td>\n",
       "      <td>8.331206e+07</td>\n",
       "      <td>11500.000000</td>\n",
       "      <td>11051.000000</td>\n",
       "      <td>10831.500000</td>\n",
       "      <td>99.500000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>7.000000</td>\n",
       "      <td>1.332661e+08</td>\n",
       "      <td>1.332714e+08</td>\n",
       "      <td>14000.000000</td>\n",
       "      <td>12577.000000</td>\n",
       "      <td>12002.000000</td>\n",
       "      <td>117.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>10.000000</td>\n",
       "      <td>1.653777e+08</td>\n",
       "      <td>1.653902e+08</td>\n",
       "      <td>20000.000000</td>\n",
       "      <td>18865.000000</td>\n",
       "      <td>17197.500000</td>\n",
       "      <td>128.500000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>10.000000</td>\n",
       "      <td>2.373608e+08</td>\n",
       "      <td>2.373808e+08</td>\n",
       "      <td>20000.000000</td>\n",
       "      <td>20004.000000</td>\n",
       "      <td>19764.000000</td>\n",
       "      <td>197.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             chr         start           end        length  n20x_1_indiv  \\\n",
       "count  15.000000  1.500000e+01  1.500000e+01     15.000000     15.000000   \n",
       "mean    6.800000  1.364851e+08  1.364995e+08  14416.000000  13667.666667   \n",
       "std     3.233751  5.854636e+07  5.854773e+07   5279.405811   4936.985411   \n",
       "min     1.000000  4.956573e+07  4.957973e+07   5340.000000   5285.000000   \n",
       "25%     5.000000  8.329806e+07  8.331206e+07  11500.000000  11051.000000   \n",
       "50%     7.000000  1.332661e+08  1.332714e+08  14000.000000  12577.000000   \n",
       "75%    10.000000  1.653777e+08  1.653902e+08  20000.000000  18865.000000   \n",
       "max    10.000000  2.373608e+08  2.373808e+08  20000.000000  20004.000000   \n",
       "\n",
       "       n20x_450_indiv       nSNVs  \n",
       "count       15.000000   15.000000  \n",
       "mean     13065.866667  122.266667  \n",
       "std       4731.881019   35.531609  \n",
       "min       5066.000000   79.000000  \n",
       "25%      10831.500000   99.500000  \n",
       "50%      12002.000000  117.000000  \n",
       "75%      17197.500000  128.500000  \n",
       "max      19764.000000  197.000000  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "targets.describe()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Target lengths and locations"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The easiest way to check for OCR errors is to check the column sums against the *Total* row of table, which says the totals are:\n",
    "- length: 216,240\n",
    "- 20x in 1 indiv: 205,015\n",
    "- 20x in 450 indivs: 195,988\n",
    "- number of snvs: 1,834"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "length: 216240\n",
      "1 indiv: 205015\n",
      "450 ind: 195988\n",
      "n snvs: 1834\n"
     ]
    }
   ],
   "source": [
    "print('length:', targets.length.sum())\n",
    "print('1 indiv:', targets.n20x_1_indiv.sum())\n",
    "print('450 ind:', targets.n20x_450_indiv.sum())\n",
    "print('n snvs:', targets.nSNVs.sum())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This checks out. There could be some subtle OCR errors, but that seems unlikely.\n",
    "\n",
    "Now, compare length to end - start:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0     True\n",
       "1     True\n",
       "2     True\n",
       "3     True\n",
       "4     True\n",
       "5     True\n",
       "6     True\n",
       "7     True\n",
       "8     True\n",
       "9     True\n",
       "10    True\n",
       "11    True\n",
       "12    True\n",
       "13    True\n",
       "14    True\n",
       "dtype: bool"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "targets.end - targets.start == targets.length"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This also checks out.\n",
    "\n",
    "The SI states that target lengths range from 5,340bp to 20,00bp."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(5340, 20000)"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(targets.length.min(), targets.length.max())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Checks out.\n",
    "\n",
    "The SI states that the targets were selected in pairs and triplets so that they are in partial LD. It says:\n",
    ">duplets or triplets cover physical distance of 195 kb on average (range: 93 kb and 430 kb)\n",
    "\n",
    "We can try to identify the duplets and triplets by eye and check this."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    chr      start        end\n",
      "0     1  237146097  237165997\n",
      "1     1  237360801  237380801\n",
      "2     4  164000000  164020000\n",
      "3     4  164115673  164135673\n",
      "4     6  165342001  165355001\n",
      "5     6  165413304  165425304\n",
      "6     6  165470500  165482500\n",
      "7     7   49565734   49579734\n",
      "8     7   49645200   49659200\n",
      "9    10   82981023   83001023\n",
      "10   10   83207715   83215715\n",
      "11   10   83388407   83408407\n",
      "12   10  133143109  133150109\n",
      "13   10  133231083  133242083\n",
      "14   10  133266053  133271393\n"
     ]
    }
   ],
   "source": [
    "print(targets[['chr', 'start', 'end']])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "These fall into 6 clear sets: Chr1, Chr4, Chr7, Chr10 around 83 Mb, and Chr10 around 133 Mb."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[234704 135673 140499  93466 427384 128284]\n",
      "Min: 93466\n",
      "Max: 427384\n",
      "Avg: 193335.0\n"
     ]
    }
   ],
   "source": [
    "group_start = [0,2,4,7,9,12]\n",
    "group_end = [1,3,6,8,11,14]\n",
    "group_lengths = np.array(targets.end[group_end]) - np.array(targets.start[group_start])\n",
    "print(group_lengths)\n",
    "print('Min:', min(group_lengths))\n",
    "print('Max:', max(group_lengths))\n",
    "print('Avg:', np.mean(group_lengths))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The minimum is correct, but the maximum and the average are both a little low. **Maybe should double check this.**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Coverage\n",
    "\n",
    "For each target, the length should be greater than the number of bases with 20x coverage in one sample, which should be greater than the number of bases with 20x coverage in 450 samples. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0     False\n",
       "1     False\n",
       "2      True\n",
       "3      True\n",
       "4      True\n",
       "5     False\n",
       "6      True\n",
       "7      True\n",
       "8      True\n",
       "9      True\n",
       "10    False\n",
       "11     True\n",
       "12    False\n",
       "13    False\n",
       "14     True\n",
       "dtype: bool"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "targets.length >= targets.n20x_1_indiv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0     True\n",
       "1     True\n",
       "2     True\n",
       "3     True\n",
       "4     True\n",
       "5     True\n",
       "6     True\n",
       "7     True\n",
       "8     True\n",
       "9     True\n",
       "10    True\n",
       "11    True\n",
       "12    True\n",
       "13    True\n",
       "14    True\n",
       "dtype: bool"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "targets.n20x_1_indiv >= targets.n20x_450_indiv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0      True\n",
       "1      True\n",
       "2      True\n",
       "3      True\n",
       "4      True\n",
       "5     False\n",
       "6      True\n",
       "7      True\n",
       "8      True\n",
       "9      True\n",
       "10     True\n",
       "11     True\n",
       "12    False\n",
       "13    False\n",
       "14     True\n",
       "dtype: bool"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "targets.length >= targets.n20x_450_indiv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The counts for 20x coverage in at least 450 individuals are all less than the bp covered at 20x in at least 1 individual, as they should be. But there are sometimes more bases covered than the target length. Presumably, this is some sort of overhang. **I should investigate and understand this.**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0       -2\n",
       "1       -4\n",
       "2     2038\n",
       "3      232\n",
       "4     1902\n",
       "5       -2\n",
       "6        0\n",
       "7     1423\n",
       "8        0\n",
       "9      128\n",
       "10      -4\n",
       "11    5466\n",
       "12      -3\n",
       "13      -4\n",
       "14      55\n",
       "dtype: int64"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "targets.length - targets.n20x_1_indiv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0      136\n",
       "1      461\n",
       "2     3883\n",
       "3      557\n",
       "4     2341\n",
       "5       -2\n",
       "6        0\n",
       "7     2817\n",
       "8      221\n",
       "9     1722\n",
       "10     622\n",
       "11    7227\n",
       "12      -3\n",
       "13      -4\n",
       "14     274\n",
       "dtype: int64"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "targets.length - targets.n20x_450_indiv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "When considering only sites with 20x coverage in 450 individuals, there are never more than 4 extra bases, so we might be ok neglecting this."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Single nucleotide variants"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We have the following information about the filtered list of SNVs:\n",
    "- there 1,834 high quality SNVs (main text p. 759, para. 1)\n",
    "- 95% of SNVs are genotyped in 450 individuals and these are the ones used in the SFS figures. (main text p. 759, para. 1)\n",
    "- the proportion of singletons among variant sites is 38.4% (main text p. 759, para. 3)\n",
    "- The Ti/Tv ratio is 2.22 (main text p. 759, para 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AN</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr7</td>\n",
       "      <td>49565781</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>984</td>\n",
       "      <td>9</td>\n",
       "      <td>0.0091</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr7</td>\n",
       "      <td>49565844</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>984</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr7</td>\n",
       "      <td>49565865</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>984</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr7</td>\n",
       "      <td>49566127</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>984</td>\n",
       "      <td>356</td>\n",
       "      <td>0.3618</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr7</td>\n",
       "      <td>49566144</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>984</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0010</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM       POS REF ALT   AN   AC      AF\n",
       "0  chr7  49565781   C   A  984    9  0.0091\n",
       "1  chr7  49565844   G   A  984    1  0.0010\n",
       "2  chr7  49565865   C   A  984    1  0.0010\n",
       "3  chr7  49566127   G   A  984  356  0.3618\n",
       "4  chr7  49566144   T   C  984    1  0.0010"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snvs = pd.read_table('../data/GazaveEtal/variants.txt', delim_whitespace=True)\n",
    "snvs.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We need to make the CHROM column match the one from the target file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>AN</th>\n",
       "      <th>AC</th>\n",
       "      <th>AF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>7</td>\n",
       "      <td>49565781</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>984</td>\n",
       "      <td>9</td>\n",
       "      <td>0.0091</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>7</td>\n",
       "      <td>49565844</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>984</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>7</td>\n",
       "      <td>49565865</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>984</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>7</td>\n",
       "      <td>49566127</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>984</td>\n",
       "      <td>356</td>\n",
       "      <td>0.3618</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>7</td>\n",
       "      <td>49566144</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>984</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0010</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM       POS REF ALT   AN   AC      AF\n",
       "0      7  49565781   C   A  984    9  0.0091\n",
       "1      7  49565844   G   A  984    1  0.0010\n",
       "2      7  49565865   C   A  984    1  0.0010\n",
       "3      7  49566127   G   A  984  356  0.3618\n",
       "4      7  49566144   T   C  984    1  0.0010"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snvs.CHROM = snvs.CHROM.apply(lambda s: int(s[3:]))\n",
    "snvs.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Number of SNPVs and genotype counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CHROM    1834\n",
       "POS      1834\n",
       "REF      1834\n",
       "ALT      1834\n",
       "AN       1834\n",
       "AC       1834\n",
       "AF       1834\n",
       "dtype: int64"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snvs.count()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "There are 1,834 SNVs in the table, suggesting that these are the \"high quality\" SNVs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "count    1834.000000\n",
       "mean      984.064340\n",
       "std         5.164633\n",
       "min       860.000000\n",
       "25%       984.000000\n",
       "50%       984.000000\n",
       "75%       984.000000\n",
       "max       986.000000\n",
       "Name: AN, dtype: float64"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snvs.AN.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<matplotlib.axes._subplots.AxesSubplot at 0x112e6c0b8>"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYEAAAD8CAYAAACRkhiPAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAFXVJREFUeJzt3X+snmd93/H3Z8nqUdwkQOiZ62Q4aAlafmyZfJqibaBj\nQZuA0BLQxBxlEA+GoaSoVJGK004CFVniV4uGGOnMEhFEl7OUwIgSshEQBzZpJrVpwHYgxSFm8Wma\njAJhpp2Hw3d/PLfLg31+5Tkn53luXe+X9Ojcz3Vf131/Lx/f/vj+8ZyTqkKS1Ka/Ne4CJEnjYwhI\nUsMMAUlqmCEgSQ0zBCSpYYaAJDXMEJCkhhkCktQwQ0CSGnbmuAtYzrnnnltbtmwZaeyPfvQjnv3s\nZ69tQeuo7/VD/+dg/ePV9/phfHPYv3//d6vq+cv1m/gQ2LJlC/v27Rtp7NzcHDMzM2tb0Drqe/3Q\n/zlY/3j1vX4Y3xySfGcl/bwcJEkNMwQkqWGGgCQ1zBCQpIYZApLUMENAkhpmCEhSwwwBSWqYISBJ\nDTMEJGmCbNl1D1t23bNu+zMEJKlhhoAkNcwQkKSGGQKS1DBDQJIatmwIJLk1yRNJDg61/eckD3Sv\nI0ke6Nq3JPnroXV/ODRma5IDSQ4n+VCSPDNTkiSt1Ep+qczHgA8DHz/ZUFX/8uRykt8Hnhzq/3BV\nXb7Adm4G3gR8BfgscBVw79MvWZK0VpY9E6iqLwPfW2hd97/51wK3L7WNJJuAs6pqb1UVg0C55umX\nK0laSxn8m7xMp2QLcHdVXXpK+0uBP6iq6aF+h4BvMTg7+LdV9d+TTAPvqaqXd/1eAryjql61yP52\nAjsBpqamts7Ozo4yN44dO8bGjRtHGjsJ+l4/9H8O1j9efa8fnv4cDswPLqxctvnsVe1327Zt+0/+\n27ykqlr2BWwBDi7QfjNw49D7DcDzuuWtwKPAWcA08Pmhfi9hECrL7nvr1q01qi9+8Ysjj50Efa+/\nqv9zsP7x6nv9VU9/Di94x931gnfcver9AvtqBf/GjvyL5pOcCbym+8f+ZKAcB453y/uTPAxcBMwD\n5w0NP69rkySN0WoeEX058M2qOnqyIcnzk5zRLb8QuBD4dlU9BvwwyYu7+wivBz6zin1LktbASh4R\nvR34n8CLkhxN8sZu1XZOvyH8UuDr3SOjnwTeUlUnbyq/FfiPwGHgYXwySJLGbtnLQVV17SLtOxZo\nuxO4c5H++4BLF1onSRoPPzEsSQ0zBCSpYYaAJDXMEJCkhhkCktQwQ0CSGmYISFLDDAFJapghIEkN\nMwQkqWGGgCQ1zBCQpIYZApLUMENAkhpmCEhSwwwBSWqYISBJDTMEJKlhhoAkNWwlv2j+1iRPJDk4\n1PauJPNJHuherxxad1OSw0keSnLlUPvWJAe6dR9KkrWfjiTp6VjJmcDHgKsWaP9gVV3evT4LkORi\nYDtwSTfmI0nO6PrfDLwJuLB7LbRNSdI6WjYEqurLwPdWuL2rgdmqOl5VjwCHgSuSbALOqqq9VVXA\nx4FrRi1akrQ2zlzF2LcleT2wD7ixqr4PbAb2DvU52rX9uFs+tX1BSXYCOwGmpqaYm5sbqcBjx46N\nPHYS9L1+6P8crH+8+l4/PP053HjZCYB1m/eoIXAz8G6guq+/D7xhrYqqqj3AHoDp6emamZkZaTtz\nc3OMOnYS9L1+6P8crH+8+l4/PP057Nh1DwBHrlv5mNUY6emgqnq8qp6qqp8AHwWu6FbNA+cPdT2v\na5vvlk9tlySN0Ugh0F3jP+nVwMknh+4CtifZkOQCBjeA76+qx4AfJnlx91TQ64HPrKJuSdIaWPZy\nUJLbgRng3CRHgXcCM0kuZ3A56AjwZoCqOpTkDuBB4ARwQ1U91W3qrQyeNHoWcG/3kiSN0bIhUFXX\nLtB8yxL9dwO7F2jfB1z6tKqTpAZs6e4DjIOfGJakhhkCktQwQ0CSGmYISFLDDAFJapghIEkNMwQk\nqWGGgCQ1zBCQpIYZApLUMENAkhpmCEhSwwwBSWqYISBJDTMEJKlhhoAkNcwQkKSGGQKS1DBDQJIa\ntmwIJLk1yRNJDg61vT/JN5N8Pcmnk5zTtW9J8tdJHuhefzg0ZmuSA0kOJ/lQkjwzU5IkrdRKzgQ+\nBlx1Stt9wKVV9Q+BPwNuGlr3cFVd3r3eMtR+M/Am4MLudeo2JUnrbNkQqKovA987pe1zVXWie7sX\nOG+pbSTZBJxVVXurqoCPA9eMVrIkaa2sxT2BNwD3Dr2/oLsU9KUkL+naNgNHh/oc7dokSWOUwX/M\nl+mUbAHurqpLT2n/XWAaeE1VVZINwMaq+sskW4H/AlwCXAS8p6pe3o17CfCOqnrVIvvbCewEmJqa\n2jo7OzvS5I4dO8bGjRtHGjsJ+l4/9H8O1j9efa8fVjaHA/NPntZ22eazV7Xfbdu27a+q6eX6nTnq\nDpLsAF4FvKy7xENVHQeOd8v7kzzMIADm+dlLRud1bQuqqj3AHoDp6emamZkZqca5uTlGHTsJ+l4/\n9H8O1j9efa8fVjaHHbvuOa3tyHVLj1krI10OSnIV8NvAP6+qvxpqf36SM7rlFzK4AfztqnoM+GGS\nF3dPBb0e+Myqq5ckrcqyZwJJbgdmgHOTHAXeyeBpoA3Afd2Tnnu7J4FeCvxekh8DPwHeUlUnbyq/\nlcGTRs9icA9h+D6CJGkMlg2Bqrp2geZbFul7J3DnIuv2AZcutE6SNB5+YliSGmYISFLDDAFJapgh\nIEkNMwQkqWGGgCQ1zBCQpIYZApLUMENAkhpmCEhSwwwBSWqYISBJDTMEJKlhhoAkNcwQkKSGGQKS\n1DBDQJIaZghIUsMMAUlq2LIhkOTWJE8kOTjU9twk9yX5Vvf1OUPrbkpyOMlDSa4cat+a5EC37kPp\nfkO9JGl8VnIm8DHgqlPadgFfqKoLgS9070lyMbAduKQb85EkZ3RjbgbeBFzYvU7dpiRpnS0bAlX1\nZeB7pzRfDdzWLd8GXDPUPltVx6vqEeAwcEWSTcBZVbW3qgr4+NAYSdKYjHpPYKqqHuuW/wKY6pY3\nA48O9TvatW3ulk9tlySN0Zmr3UBVVZJai2JOSrIT2AkwNTXF3NzcSNs5duzYyGMnQd/rh/7PwfrH\nq+/1w8rmcONlJ05rW695jxoCjyfZVFWPdZd6nuja54Hzh/qd17XNd8unti+oqvYAewCmp6drZmZm\npCLn5uYYdewk6Hv90P85WP949b1+WNkcduy657S2I9ctPWatjHo56C7g+m75euAzQ+3bk2xIcgGD\nG8D3d5eOfpjkxd1TQa8fGiNJGpNlzwSS3A7MAOcmOQq8E3gPcEeSNwLfAV4LUFWHktwBPAicAG6o\nqqe6Tb2VwZNGzwLu7V6SpDFaNgSq6tpFVr1skf67gd0LtO8DLn1a1UmSnlF+YliSGmYISFLDDAFJ\napghIEkNMwQkqWGGgCQ1zBCQpIYZApLUMENAkhpmCEhSwwwBSWqYISBJDTMEJKlhhoAkNcwQkKSG\nGQKS1DBDQJIaZghIUsMMAUlq2MghkORFSR4Yev0wyduTvCvJ/FD7K4fG3JTkcJKHkly5NlOQJI1q\n2V80v5iqegi4HCDJGcA88GngXwMfrKoPDPdPcjGwHbgE+CXg80kuqqqnRq1BkrQ6a3U56GXAw1X1\nnSX6XA3MVtXxqnoEOAxcsUb7lySNYK1CYDtw+9D7tyX5epJbkzyna9sMPDrU52jXJkkak1TV6jaQ\n/Bzw58AlVfV4kingu0AB7wY2VdUbknwY2FtVn+jG3QLcW1WfXGCbO4GdAFNTU1tnZ2dHqu3YsWNs\n3LhxpLGToO/1Q//nYP3j1ff6YWVzODD/5Gltl20+e1X73bZt2/6qml6u38j3BIa8AvhqVT0OcPIr\nQJKPAnd3b+eB84fGnde1naaq9gB7AKanp2tmZmakwubm5hh17CToe/3Q/zlY/3j1vX5Y2Rx27Lrn\ntLYj1y09Zq2sxeWgaxm6FJRk09C6VwMHu+W7gO1JNiS5ALgQuH8N9i9JGtGqzgSSPBv4VeDNQ83v\nS3I5g8tBR06uq6pDSe4AHgROADf4ZJAkjdeqQqCqfgQ875S21y3RfzewezX7lCStHT8xLEkNMwQk\nqWGGgCQ1zBCQpIYZApLUMENAkhpmCEhSwwwBSWqYISBJDTMEJKlhhoAkNcwQkKSGGQKS1DBDQJIa\nZghIUsMMAUlqmCEgSQ0zBCSpYYaAJDXMEJCkhq0qBJIcSXIgyQNJ9nVtz01yX5JvdV+fM9T/piSH\nkzyU5MrVFi9JWp21OBPYVlWXV9V0934X8IWquhD4QveeJBcD24FLgKuAjyQ5Yw32L0ka0TNxOehq\n4LZu+TbgmqH22ao6XlWPAIeBK56B/UuSVihVNfrg5BHgSeAp4D9U1Z4kP6iqc7r1Ab5fVeck+TCw\nt6o+0a27Bbi3qj65wHZ3AjsBpqamts7Ozo5U37Fjx9i4ceNIYydB3+uH/s/B+ser7/XDyuZwYP7J\n09ou23z2qva7bdu2/UNXaBZ15qr2Av+squaT/CJwX5JvDq+sqkrytFOmqvYAewCmp6drZmZmpOLm\n5uYYdewk6Hv90P85WP949b1+WNkcduy657S2I9ctPWatrOpyUFXNd1+fAD7N4PLO40k2AXRfn+i6\nzwPnDw0/r2uTJI3JyCGQ5NlJfuHkMvBrwEHgLuD6rtv1wGe65buA7Uk2JLkAuBC4f9T9S5JWbzWX\ng6aATw8u+3Mm8J+q6r8m+RPgjiRvBL4DvBagqg4luQN4EDgB3FBVT62qeknSqowcAlX1beAfLdD+\nl8DLFhmzG9g96j4lSWvLTwxLUsMMAUlqmCEgSQ0zBCSpYYaAJDXMEJCkhhkCktQwQ0CSGmYISFLD\nDAFJapghIEkNMwQkqWGGgCQ1zBCQpIYZApLUMENAkhpmCEhSwwwBSWqYISBJDRs5BJKcn+SLSR5M\ncijJb3bt70oyn+SB7vXKoTE3JTmc5KEkV67FBCRJoxv5F80DJ4Abq+qrSX4B2J/kvm7dB6vqA8Od\nk1wMbAcuAX4J+HySi6rqqVXUIElahZHPBKrqsar6arf8f4BvAJuXGHI1MFtVx6vqEeAwcMWo+5ck\nrd6a3BNIsgX4x8BXuqa3Jfl6kluTPKdr2ww8OjTsKEuHhiTpGZaqWt0Gko3Al4DdVfWpJFPAd4EC\n3g1sqqo3JPkwsLeqPtGNuwW4t6o+ucA2dwI7AaamprbOzs6OVNuxY8fYuHHjSGMnQd/rh/7PwfrH\nq+/1w8rmcGD+ydPaLtt89qr2u23btv1VNb1cv9XcEyDJ3wbuBP6oqj4FUFWPD63/KHB393YeOH9o\n+Hld22mqag+wB2B6erpmZmZGqm9ubo5Rx06CvtcP/Z+D9Y9X3+uHlc1hx657Tms7ct3SY9bKap4O\nCnAL8I2q+oOh9k1D3V4NHOyW7wK2J9mQ5ALgQuD+UfcvSVq91ZwJ/FPgdcCBJA90bb8DXJvkcgaX\ng44AbwaoqkNJ7gAeZPBk0Q0+GSRJ4zVyCFTV/wCywKrPLjFmN7B71H1KktaWnxiWpIYZApLUMENA\nkhpmCEhSwwwBSWqYISBJDTMEJKlhhoAkNcwQkKSGGQKS1DBDQJIaZghIUsMMAUlqmCEgSQ0zBCSp\nYYaAJDXMEJCkhhkCktSw1fyOYUnSCLbsumfcJfyNdT8TSHJVkoeSHE6ya733L0n6qXUNgSRnAP8e\neAVwMXBtkovXswZJ0k+t9+WgK4DDVfVtgCSzwNXAg+tchyStykov6dx42Ql2TNDln1OtdwhsBh4d\nen8U+JV1rkFLWOtrlZN+ACzH+ser7/X3Qapq/XaW/Avgqqr6N9371wG/UlW/cUq/ncDO7u2LgIdG\n3OW5wHdHHDsJ+l4/9H8O1j9efa8fxjeHF1TV85frtN5nAvPA+UPvz+vafkZV7QH2rHZnSfZV1fRq\ntzMufa8f+j8H6x+vvtcPkz+H9X466E+AC5NckOTngO3AXetcgySps65nAlV1IslvAP8NOAO4taoO\nrWcNkqSfWvcPi1XVZ4HPrtPuVn1Jacz6Xj/0fw7WP159rx8mfA7remNYkjRZ/NlBktSwXodAkt9K\ncijJwSS3J/k7XfvbknyzW/e+of43dT+u4qEkV46v8r+p57T6k1yeZG+SB5LsS3LFUP+Jqh8gyW92\n9R9K8vau7blJ7kvyre7rc4b6T9QcFqn//d3fn68n+XSSc4b6T3z9Q+tuTFJJzh1qm6j6YfE59Og4\nXujvUH+O46rq5YvBB88eAZ7Vvb8D2AFsAz4PbOjaf7H7ejHwNWADcAHwMHDGBNb/OeAVXdsrgblJ\nrL+r6VLgIPDzDO4vfR74+8D7gF1dn13AeydxDkvU/2vAmV2f9/at/m7d+QwewPgOcO4k1r/M96Av\nx/Fi9ffmOO71mQCDP/RnJTmTwTfhz4FfB95TVccBquqJru/VwGxVHa+qR4DDDH6MxTgtVH8BZ3Xr\nz+7aYDLr/wfAV6rqr6rqBPAl4DUMar2t63MbcE23PGlzWLD+qvpc9x5gL4PPs0BP6u/WfRD4bQZ/\nn06atPph8Tn05TherP7eHMe9DYGqmgc+APwv4DHgyar6HHAR8JIkX0nypSS/3A1Z6EdWbF7Pmoct\nUf/bgfcnebRbf1M3ZKLq7xxk8Gf9vCQ/z+B/POcDU1X1WNfnL4CpbnnS5rBY/cPeANzbLfei/iRX\nA/NV9bVT+k9a/bD496AXxzGL19+b47i3v0+gu858NYNTqh8Af5zkXzGY03OBFwO/DNyR5IVjK3QR\nS9R/BfBbVXVnktcCtwAvH1+li6uqbyR5L4NT3x8BDwBPndKnkkzkI2jL1Z/kd4ETwB+Np8KlLVL/\nBuB3GFzSmnhLfA96cRwvUf+v05PjuLdnAgz+QB+pqv9dVT8GPgX8EwbJ+qkauB/4CYOf3bGiH1mx\njhar//puGeCP+emp4qTVD0BV3VJVW6vqpcD3gT8DHk+yCaD7evJUfuLmsEj9JNkBvAq4rrqLufSj\n/kMM/mPxtSRHGNT41SR/lwmsHxb9HvTlOF6s/v4cx+O8IbGaF4OfPnqIwbX0MLj2/DbgLcDvdX0u\nYnDqFeASfvaGzLcZ7w2lxer/BjDT9XkZsL9bnqj6h+Zx8obd3wO+CZwDvJ+fvTH8vkmdwyL1X8Xg\nx5s//5S+vaj/lPVH+OmN4Ymrf4nvQS+O4yXq781x3NvLQVX1lSSfBL7K4JT9Txl8Mq+AW5McBP4f\ncH0N/vQPJbmDwcF9Arihqp5aeOvPvCXq/1Pg33U3i/8v3U9TraqJqn/InUmeB/yYQU0/SPIeBqfv\nb2TwdMprYWLnsFD9H2ZwkN6XBGBvVb2lL/Uv1nFC64eFvwe30oPjuLNQ/W+iJ8exnxiWpIb1+Z6A\nJGmVDAFJapghIEkNMwQkqWGGgCQ1zBCQpIYZApLUMENAkhr2/wHysYF+hVSdAQAAAABJRU5ErkJg\ngg==\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x112e6cf28>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "snvs.AN.hist(cumulative=True, bins=np.arange(860,987,1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SNVs with fewer than 450 genotypes 3\n",
      "Fraction with at least 450 genotypes 0.998364231189\n"
     ]
    }
   ],
   "source": [
    "print('SNVs with fewer than 450 genotypes', (snvs.AN < 900).sum())\n",
    "print('Fraction with at least 450 genotypes', (snvs.AN >= 900).sum() / snvs.AN.count())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The proportion genotyped in at least 450 individuals is much larger than 95%. This suggests that this table was filtered after the the 95% number was calculated. On the other hand, the table cannot have been filtered on genotypes being greater than or equal to 900, because a few SNVs don't meet this criterion. This finding is potentially problematic, because it means that some of the \"unobserved\" sites are known to have variants. On the other hand, the number of variants is the same as reported in the main text, so something is odd here.\n",
    "\n",
    "One thing to check is that the AN field of the .vcf really corresponds to number of genotypes called. If this is the case, then AF should equal AC/AN:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "count    1834.000000\n",
       "mean        0.000009\n",
       "std         0.000024\n",
       "min        -0.000050\n",
       "25%        -0.000008\n",
       "50%         0.000016\n",
       "75%         0.000020\n",
       "max         0.000050\n",
       "dtype: float64"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "freqs = snvs.AC / snvs.AN\n",
    "(freqs - snvs.AF).describe()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The differences are all in the fifth decimal place, while the AF column is only recorded to four places, suggesting that the AF = AC/AN up to a rounding error."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Singleton fraction"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.37786259541984735"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(snvs.AC == 1).sum() / snvs.AC.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.3784817039868924"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "((snvs.AC == 1) & (snvs.AN >= 900)).sum() / (snvs.AN >= 900).sum()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The fraction of singletons is close to but not exactly 38.4%. This is true regardless of whether we filter on having 450 genotyped samples at the site. I see three possiblities:\n",
    "1. The 38.4% number refers to the downsampled frequencies. (I'll test this below.)\n",
    "2. The data file I've downloaded contains data that was collected after the paper was published.\n",
    "3. Filters were applied to individual genotypes *after* generating the file. I can probably filter the genotypes myself from the vcf file using `bcftools`.\n",
    "\n",
    "The relevant passage from SI p.9 on filtering individual genotypes:\n",
    "> Individual genotypes were marked as missing if they presented strong allelic imbalance (> 20%), a depth of coverage < 20x or a low individual genotype quality (QC < 30), resulting in only a small amount of missing data. We established that 95% of SNVs have successful calls for at least 900 individual chromosomes out of the possible 986."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Transitions/Transversions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2.2231985940246046"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "base_table = {'A':'R', 'G':'R', 'C':'Y', 'T':'Y'}\n",
    "snvs['is_TI'] = snvs.REF.replace(base_table) == snvs.ALT.replace(base_table)\n",
    "snvs.is_TI.mean() / (1 - snvs.is_TI.mean())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The Ti/Tv ratio is 2.22, as reported."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## SNV locations"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Here we want to check whether all the SNVs are within target regions. If so, we might be able to ignore the potential overhangs reported above in the Target Regions section."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "for index, row in targets.iterrows():\n",
    "    snvs['in_target{}'.format(index)] = (snvs.CHROM==row.chr) & (snvs.POS >= row.start) & (snvs.POS < row.end)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "count    1834.0\n",
       "mean        1.0\n",
       "std         0.0\n",
       "min         1.0\n",
       "25%         1.0\n",
       "50%         1.0\n",
       "75%         1.0\n",
       "max         1.0\n",
       "dtype: float64"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snvs[['in_target{}'.format(i) for i in range(len(targets))]].sum(axis=1).describe()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "All snvs are within exactly one target."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CHROM                                                      11552\n",
       "POS                                                 259402467771\n",
       "REF            CGCGTGGCGATGTGCGAACACAACGAGGCAGGCGAGAAATAATGGT...\n",
       "ALT            AAAACAAACTCACATACGATTGTAAGAAATAATTGAGCGCGCGAAC...\n",
       "AN                                                       1804774\n",
       "AC                                                        195291\n",
       "AF                                                       198.518\n",
       "is_TI                                                       1265\n",
       "in_target0                                                   180\n",
       "in_target1                                                   197\n",
       "in_target2                                                    96\n",
       "in_target3                                                   119\n",
       "in_target4                                                    86\n",
       "in_target5                                                   103\n",
       "in_target6                                                   122\n",
       "in_target7                                                   115\n",
       "in_target8                                                   172\n",
       "in_target9                                                   126\n",
       "in_target10                                                   79\n",
       "in_target11                                                  131\n",
       "in_target12                                                   81\n",
       "in_target13                                                  117\n",
       "in_target14                                                  110\n",
       "dtype: object"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snvs.sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0     180\n",
       "1     197\n",
       "2      96\n",
       "3     119\n",
       "4      86\n",
       "5     103\n",
       "6     122\n",
       "7     115\n",
       "8     172\n",
       "9     126\n",
       "10     79\n",
       "11    131\n",
       "12     81\n",
       "13    117\n",
       "14    110\n",
       "Name: nSNVs, dtype: int64"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "targets.nSNVs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "All targets have the right number of SNVs."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Site frequency spectrum\n",
    "The SFS for figures is calculated according to a downsampling and binning procedure described on SI P. 23:\n",
    "- Remove SNVs with fewer than 900 genotyped chromosomes.\n",
    "- Downsample the remaining SNVs to 900 chromosomes according to a hypergeometric distribution.\n",
    "- Calculate the SFS as a histogram with bins: (1, 2, 3, 4, 5, 6-10, 11-20, 21-50, 51-100, 101-200, 201-450).\n",
    "- Normalize counts by width of each bin."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Define bin edges according to np.hist specification. Including 451 in the last bin to make sure the bin widths are correct.\n",
    "bins = np.array([1, 2, 3, 4, 5, 6, 11, 21, 51, 101, 201, 451])\n",
    "bin_widths = np.diff(bins)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "from scipy.stats import hypergeom\n",
    "n_ds = 900\n",
    "counts = np.arange(n_ds+1, dtype=int)\n",
    "#ds_cdf = hypergeom(snvs.AN[snvs.AN>=900], n_ds, snvs.AC[snvs.AN>=900]).cdf(bins[:,None])\n",
    "ds_pmf = hypergeom(snvs.AN, n_ds, snvs.AC).pmf(counts[:,None])\n",
    "ds_cdf = np.cumsum(ds_pmf, axis=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "binned_pmf = np.diff(ds_cdf[bins-1,:], axis=0)\n",
    "binned_sfs = np.nansum(binned_pmf, axis=1) / bin_widths\n",
    "binned_sfs /= (snvs.AN >= 900).sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[<matplotlib.lines.Line2D at 0x11a5fb198>]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYAAAAEACAYAAAC6d6FnAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAHxdJREFUeJzt3Wl4VdWhxvH/ykRICIchYcxAIAGZpzAJ4oBWHBBnQapW\nUbTi1NZrtdd76/W2VXs7KApSqlStKCIqgmOtyoxAmJEpIQxJGBIChEBCxnU/EBBRICHD3ufs9/c8\n+XA2Oycvz37gzTpr7bWNtRYREfGeIKcDiIiIM1QAIiIepQIQEfEoFYCIiEepAEREPEoFICLiUSoA\nERGPUgGIiHiUCkBExKNUACIiHhXidIAfY4wZAYyIioq6p2PHjk7HERHxKytWrNhnrY0523nGzXsB\npaSk2NTUVKdjiIj4FWPMCmttytnO00dAIiIepQIQEfEoFYCIiEepAEREPMqVBWCMGWGMmZKfn+90\nFBGRgOXKArDWzrHWjvP5fE5HEREJWK4sgJrKyD3M0ow8p2OIiLhawBWAtZZH313D+LdWkltQ7HQc\nERHXCrgCMMbwzPU9KDhaxmMz1+DmG91ERJwUcAUA0KlVFP95VWe+3pzL64u3Ox1HRMSVArIAAG4b\nmMCw81rwh083sWnPIafjiIi4TsAWgDGG527sQePwUB56exVHS8udjiQi4ioBWwAA0Y0a8Oebe7Jl\n72Ge+WSj03FERFzFlQVQmzeCXdgxhrFDEnl9yQ6+2rS3FtKJiAQGVxZAbd8I9tjwTnRu3ZhH311L\nTsHRWnlPERF/58oCqG0NQoKZMKoXR4rLePTdtVRUaGmoiIgnCgAguWUU/3V1F+ZvyeUfWhoqIuKd\nAgAYMyCey7q05LlPN/HtLm00JyLe5qkCMMbw3A09aBIRysPTV1NUoqWhIuJdnioAgGaRYfz55p6k\n5xzm959scDqOiIhjPFcAABckxzBuaHve/GYnX2zQ0lAR8SZPFgDAoz/pRNc2jXls5hr2HtLSUBHx\nHs8WQFhIEC+M6k1RaTm/mrFGS0NFxHM8WwAASS0a8dsRXVmYvo9XF25zOo6ISL3ydAEAjOoXx+Vd\nW/LHzzexPltLQ0XEO1xZAPX5UHhjDM9e34PmkQ14aPoqCkvK6vxnioi4gSsLoL4fCt80Moy/3NyT\nbfuO8L8faddQEfEGVxaAE85PiubeoR14e9lOPlu/x+k4IiJ1TgVwkl9e1pHubX08/v5a9uRraaiI\nBDYVwEmOLQ3tRXFpBb+csVpLQ0UkoKkATtE+phFPXdOFxVvzmLIgw+k4IiJ1RgXwI25OiePK7q34\n0+ebWZt10Ok4IiJ1QgXwI4wxPHNdD2KiGvDw9NUcKdbSUBEJPCqA0/BFhPLXW3qxPe8IT8/RrqEi\nEnhUAGcwsH1zfn5hB95JzeSTdbudjiMiUqtUAGfxi8s60jPWx+PvrWXXwSKn44iI1BoVwFmEBh/b\nNbS8wvKLd1ZTrqWhIhIgVABV0C46kqeu6crSbfuZPG+r03FERGqFCqCKbuwby1U9WvPXL7awOlNL\nQ0XE/6kAqsgYwx+u7U7LxuE8PH0Vh7U0VET8nAqgGo4vDc3cX8hTs791Oo6ISI3UWwEYY9obY141\nxsysr59ZF/onNmP8xUnMXJHFnDW7nI4jInLOqlQAxpipxpgcY8z6U44PN8ZsNsakG2MeP9N7WGsz\nrLVjaxLWLR4alkyvuCb85oN1ZB0odDqOiMg5qeoI4DVg+MkHjDHBwETgCqALMNoY08UY090Y89Ep\nXy1qNbXDQoODmDCqN9bCL99Zo6WhIuKXqlQA1tr5wP5TDvcH0it/sy8BpgMjrbXrrLVXn/KVU8u5\nHRffPIKnR3Zl2fb9TPo63ek4IiLVVpM5gLZA5kmvsyqP/ShjTHNjzGSgtzHmiTOcN84Yk2qMSc3N\nza1BvLp3Xe+2XNOzDc9/mcaKHQecjiMiUi31Nglsrc2z1t5nre1grX3mDOdNsdamWGtTYmJi6ive\nOTHG8LvrutHaF84j76yi4Gip05FERKqsJgWQDcSd9Dq28pinNA4P5flbepF9oIjffqiloSLiP2pS\nAMuBZGNMojEmDBgFzK6NUMaYEcaYKfn5+bXxdnUupV0zHrwkmfdXZfPhas91oIj4qaouA30bWAJ0\nMsZkGWPGWmvLgAeAz4GNwAxrba38CmytnWOtHefz+Wrj7erFg5ck0TehKU9+sJ7M/VoaKiLuV9VV\nQKOtta2ttaHW2lhr7auVxz+x1nas/Fz/93Ub1d1CgoN4/pZeADzyzmrKyiscTiQicmau3ArC3z4C\nOi6uWQS/u64bK3Yc4CUtDRURl3NlAfjjR0DHjezVlut6t2XCl2mkbj/11gkREfdwZQH4u6dHdqVt\n04Y8PH01h7Q0VERcSgVQB6LCQ3n+lt7sOXSUJz9Yj7XaKkJE3MeVBeCvcwAn65vQlIeHJTN7zS7e\nXZHldBwRkR9wZQH48xzAycZfnMSg9s35r1nr2bDrkNNxRES+x5UFECiCgwwvjO6Fr2Eo909bofkA\nEXEVFUAdaxEVzouje5N5oIhfz1yr+QARcQ0VQD0Y0L45j13eiU/X72Hqou1OxxERAVxaAIEwCXyq\ncUPbc1mXljzzyUZW7ND9ASLiPFcWQKBMAp/MGMOfbupJmyYNGT9tFXmHi52OJCIe58oCCFS+hqFM\nGtOH/YUlPPLOaj1KUkQcpQKoZ93a+nj6mq4sSNvHhC/TnI4jIh6mAnDALf3iuKFPLBO+SmPeFnc/\n9lJEApcrCyAQJ4FPZozhd9d2o1PLKB6ZvopdB4ucjiQiHuTKAgjESeBTNQwLZtKYPpSWW8a/tZKS\nMj0/QETqlysLwCvaxzTiuRt6sGrnQZ75dKPTcUTEY1QADruqR2vuHNyOfyzazsdrdzsdR0Q8RAXg\nAk9c0Zne8U14bOYatuYedjqOiHiECsAFwkKCmHhrH8JCgrj/zZUUlZQ7HUlEPEAF4BJtmjTk+VG9\n2ZJTwH/OWqdN40SkzrmyAAJ9GejpXNgxhocuSeb9ldm8szzT6TgiEuBcWQBeWAZ6Og8NS+aC5Gj+\ne/a3rM/2VgGKSP1yZQF4WXCQ4flbetEsIoz7p60kv0gPkRGRuqECcKHmjRowcUxvdh0s4tF312g+\nQETqhArApfomNOOJKzvzxYa9/H1BhtNxRCQAqQBc7K7B7biiWyue+2wzy7bpITIiUrtUAC5mjOGP\nN/YgvlkED7y1ktwCPURGRGqPCsDlosKPPUQmv6iUh95epYfIiEitcWUBePU+gNPp3Loxv7u2G0sy\n8vjrF1ucjiMiAcKVBeDl+wBO56aUOG5JieOlr9P5aO0up+OISABwZQHIj/ufkV3pm9CUB95axaS5\n6VoeKiI1ogLwI+GhwUy7ewDX9GzDHz/bzK9mrOFoqTaOE5FzE+J0AKme8NBgXhjVi+QWjfjzF1vY\nnneEv92WQkxUA6ejiYif0QjADxljeHBYMpPG9GHD7kNcO3ERG3cfcjqWiPgZFYAfu7J7a96993zK\nKyw3vLyYLzbsdTqSiPgRFYCf6x7r48MHBpPUohHj/pnK5HlbNTksIlWiAggALRuH8864QVzZvTXP\nfrqJ/5i5luIyTQ6LyJlpEjhANAwL5qXRvUlu0Yjn/53GjrwjTP5pX5o30uSwiPw4jQACiDGGRy7t\nyEu39mZtVj4jJy5i854Cp2OJiEupAALQ1T3aMOPeQZSUVXD9pEV8tUmTwyLyQ64sAO0FVHM945ow\n+4EhJMZEMvb1VF5ZkKHJYRH5HlcWgPYCqh2tfOHMuHcQV3Rrxe8+3sij766lsKTM6Vgi4hKuLACp\nPRFhIbw0ug8PD0vm/VVZXD1hIeuyNLISERWAJwQFGX5xWUfeunsghSXlXP/yIibP20qFni0g4mkq\nAA8Z1KE5nz1yAZd2bsmzn27itqlL2ZN/1OlYIuIQFYDHNIkIY9KYPjx3Q3dW7jjI8Bfm8/m3e5yO\nJSIOUAF4kDGGW/rF89FDQ4ht2pB7/7mC33ywjqIS3T0s4iUqAA/rENOI938+mHsvbM9bS3dy9YsL\nWJ+tCWIRr1ABeFxYSBBPXNGZaXcP4HBxGddNWsQrCzI0QSziAcbNNwelpKTY1NRUp2N4xoEjJfz6\nvbX8a8NeesU1YUhSNN3a+uge66ONLxxjjNMRRaQKjDErrLUpZztPm8HJCU0jw/jbbX2ZvjyT1xdv\n5+V5WymvHAk0jwyjW1sfPWJ9x0qhrY/WKgURv6YRgJzW0dJyNuw+xPrsfNZm5bM+O5+0nMMnSiG6\nUWUptPVVlkMTWjZuoFIQcZhGAFJj4aHB9IlvSp/4pieOFZX8sBTmb8nl+JRBdKMG3xsl9Ij10bJx\nuEN/AxE5ExWAVEvDsGD6JjSlb8J3pVBYUsbG3YdYl5XP2uxjpTB3c86JUoiJanDSKOFYMbRQKYg4\nTgUgNRYRFkLfhGb0TWh24lhhSRkbdh06MUpYl53PV5tzOP6JY8vGDeh+Uil0a+ujRZRKQaQ+qQCk\nTkSEhZDSrhkp7b4rhSPFZWzY/V0prM06yJebviuFVo3DvzdK6NbWR0yUnmgmUlfqrQCMMdcCVwGN\ngVettf+qr58t7hDZIIR+7ZrR76RSOFx8fKRw8FgpZOfz5aa9J0qhtS+c+y9O4raBCQ6lFglcVSoA\nY8xU4Gogx1rb7aTjw4EXgGDgFWvts6d7D2vtLGCWMaYp8CdABSA0ahBC/8Rm9E/8fil8W/mx0Rcb\n9vJfs9azZU8B/z2iC6HBundRpLZUaRmoMWYocBh443gBGGOCgS3AZUAWsBwYzbEyeOaUt7jLWptT\n+X1/BqZZa1ee7edqGaiUV1j++Nkm/jY/g8FJzZl0a198EaFOxxJxtaouA63Sr1PW2vnA/lMO9wfS\nrbUZ1toSYDow0lq7zlp79SlfOeaY54BPq/KfvwhAcJDhiSs783839mDZtv1cN2kRGbmHnY4lEhBq\nMp5uC2Se9Dqr8tjpPAhcCtxojLnvdCcZY8YZY1KNMam5ubk1iCeB5KaUON66ZyAHi0q5duIiFqXv\nczqSiN+rtw9UrbUTrLV9rbX3WWsnn+G8KdbaFGttSkxMTH3FEz/Qr10zPhw/mFa+cG6fuow3v9nh\ndCQRv1aTAsgG4k56HVt5TKTOxDWL4L2fn8/Q5GienLWep2Z/S1l5hdOxRPxSTQpgOZBsjEk0xoQB\no4DZtRHKGDPCGDMlP19708sPRYWH8sod/bh7SCKvLd7Ona8tJ7+o1OlYIn6nSgVgjHkbWAJ0MsZk\nGWPGWmvLgAeAz4GNwAxr7be1EcpaO8daO87n89XG20kACg4yPHl1F569vjtLtuZx/aRFbN93xOlY\nIn5Fu4GK31uyNY+fT1sBwMtj+jKoQ3OHE4k4q1aXgYq42aAOzflw/GCaR4Zx26tLmb5sp9ORRPyC\nKwtAcwBSXQnNI/lg/GDOT4rm8ffX8fScDSeeWyAiP86VBaA5ADkXjcNDmXpHCj87vx1TF21j7OvL\nKTiqyWGR03FlAYicq5DgIJ66piu/v64bC9P2cf2kxezMK3Q6logrqQAkII0ZkMAbd/Unp6CYayct\nYtm2U3cyERFXFoDmAKQ2nJ8Uzazxg2nSMJQxr3zDjNTMs3+TiIe4sgA0ByC1JTE6kg/uH8yAxOY8\nNnMtf/hkoyaHRSq5sgBEapMvIpR/3NmP2wYmMGV+BuPeSOVwcZnTsUQcpwIQTwgNDuJ/r+3G0yO7\nMndLLjdMWkzmfk0Oi7e5sgA0ByB15fZB7Xjtzn7syi/i2omLSN2uyWHxLlcWgOYApC5dkBzDB/cP\nJio8hFv/vpT3VmQ5HUnEEa4sAJG6ltSiEbPGD6ZvQlN+9e4anvtsExWaHBaPUQGIZzWJCOONsf0Z\n3T+el+du5d43V3BEk8PiISoA8bTQ4CD+cF03fjuiC19u3MuNk5eQfbDI6Vgi9UIFIJ5njOHOwYlM\n/Vk/svYXMvKlRazcecDpWCJ1zpUFoFVA4oSLOrXg/fvPJyIsmFFTvuHD1XrCqQQ2VxaAVgGJU5Jb\nRjFr/GB6xTXh4emr+dPnmzU5LAHLlQUg4qRmkWG8OXYAN6fE8tLX6dw/bSWFJZoclsCjAhD5EWEh\nQTx3Qw+evKozn2/Yw02Tl7A7X5PDElhUACKnYYzh7gva8+odKWzfd4SrJizk7/MzNBqQgKECEDmL\nS85ryQfjB3Neqyh+/8lGLnjuaybP26p7BsTvGWvdN8FljBkBjEhKSronLS3N6TgiJyzfvp8JX6ax\nIG0fTSNCufuC9tw+KIGo8FCno4mcYIxZYa1NOet5biyA41JSUmxqaqrTMUR+YOXOA7z4ZRpfb87F\n1zCUsUMSueP8dvgaqgjEeSoAkXqwNusgE75M498bc4gKD+GuwYncNTgRX4SKQJyjAhCpR+uz83nx\nqzQ+/3YvjRqE8LPz2zF2SCJNI8OcjiYepAIQccCGXYd46es0Plm3h8iwYG4/vx13D0mkeaMGTkcT\nD1EBiDho854CXvo6nY/W7iI8JJjbBiVwzwXtiYlSEUjdUwGIuEB6TgEvfZXO7DW7CAsJYsyABO4d\n2p4WjcOdjiYBTAUg4iIZuYeZ+PVWZq3OJjjIcGv/eO67sAOtfCoCqX0qABEX2pF3hElfb+W9lVkE\nGcMt/eK476IOtG3S0OloEkD8ugB0I5gEusz9hUyau5WZKzIBuLFvHPdf1IG4ZhEOJ5NA4NcFcJxG\nABLosg8W8fLcdGYsz6LCWm7oE8v4i5OIb64ikHOnAhDxI7vzi/jbvAzeWraT8grLdb3bMv7iJBKj\nI52OJn5IBSDih3IOHWXyvAymLd1BaXkFI3sdK4KkFo2cjiZ+RAUg4sdyCo7yyoJt/HPJDo6WlTOi\nRxsevCSJ5JZRTkcTP6ACEAkAeYeLeWXhNt5YvJ3C0nKu7NaaB4clcV6rxk5HExdTAYgEkANHSnh1\n4TZeW7ydw8VlDO/aigeHJdG1jZ6bLT+kAhAJQPmFpUxdtI2pi7ZRcLSMSzu35OFhyXSPVRHId1QA\nIgEsv6iU1xdv59WF28gvKuWS81rw0LBkesU1cTqauIAKQMQDCo6W8saSHbyyIIMDhaUM7RjDw8OS\n6ZvQ1Olo4iAVgIiHHC4u481vdvD3+RnkHSlhSFI0Dw1Lpn9iM6ejiQNUACIeVFhSxltLdzJ5Xgb7\nDhczsH0z7hjUjsHJ0TTWc4s9w68LQHsBidTM0dLyyiLYSk5BMcFBhj7xTRiaHMPQjjF0a+sjOMg4\nHVPqiF8XwHEaAYjUTGl5BSt3HGB+Wi7zt+xjXXY+AE0jQhmSHMPQ5GiGdoyhpZ5PEFBUACLyA3mH\ni1mYvo95W3JZkLaP3IJiAM5rFcXQjjEMTY4hpV1TwkODHU4qNaECEJEzstaycXdB5eggl9TtBygp\nryA8NIiB7Zuf+LioQ0wkxujjIn+iAhCRaiksKeObjDzmb9nH/C25ZOw7AkDbJg0Z2jGaockxnJ8U\nja+hJpPdTgUgIjWSub/wxOhgcXoeBcVlBAcZesUdn0yOpkdsE00mu5AKQERqTWl5BaszDzJ/y7FC\nWJudj7XQJCKUwUnRXFj5cZGecewOKgARqTP7j5SwMH3fiULIqZxM7tiy0Ym5g/6JzTSZ7BAVgIjU\nC2stm/cWVJbBPpZt209JeQUNQionkzvGcGHHaDrENNJkcj1RAYiII4pKyvlmWx7zNucyPy2XjNzv\nJpPvu6gDo/rFERoc5HDKwKYCEBFXyDpQyIK0fby/Movl2w/QPjqSx4Z34vKurTQiqCMqABFxFWst\nX27M4dnPNpGec5g+8U144srO9GunDetqW1ULQOMwEakXxhgu7dKSzx6+gGev7072wSJumryEu19P\nJT2nwOl4nqQRgIg4oqiknKmLtvHy3K0UlpRxc0ocv7iso/YlqgX6CEhE/ELe4WJe/CqdaUt3EBxk\nuHtIe+69sD1R2r76nKkARMSv7Mwr5P/+tZk5a3bRLDKMBy9JYsyABMJC9El1dbluDsAY09kYM9kY\nM9MY8/P6+rki4h/im0fw4ujezH5gMOe1iuJ/5mzg0r/MY/aaXVRUuPcXVX9WpQIwxkw1xuQYY9af\ncny4MWazMSbdGPP4md7DWrvRWnsfcDMw+Nwji0gg6xHbhGl3D+C1O/sRERbMQ2+vYuTERSxO3+d0\ntIBT1RHAa8Dwkw8YY4KBicAVQBdgtDGmizGmuzHmo1O+WlR+zzXAx8AntfY3EJGAY4zhok4t+Pih\nC/jTTT3JO1zMra8s5Wf/WMbG3YecjhcwqjwHYIxpB3xkre1W+XoQ8JS19vLK108AWGufqcJ7fWyt\nveps52kOQETg2CMuX1+8nYlfp1NQXMb1vWP55U860rZJQ6ejuVJV5wBCavAz2gKZJ73OAgacIdBF\nwPVAA84wAjDGjAPGAcTHx9cgnogEivDQYO69sAO39Itj0tytvLZ4O3PW7uLO89tx/0VJ+CK0Yuhc\n1KQAqsVaOxeYW4XzpgBT4NgIoG5TiYg/aRIRxm+u7MztgxL4yxdbmLIgg+nLMxl/cQduH9ROu49W\nU01WAWUDcSe9jq08JiJSp2KbRvCXm3vx8YMX0DOuCX/4ZBPD/jyP91dmacVQNdSkAJYDycaYRGNM\nGDAKmF0boYwxI4wxU/Lz82vj7UQkQHVp05g37urPtLsH0DQylF/OWMNVLy5k3pZc3HyPk1tUdRno\n28ASoJMxJssYM9ZaWwY8AHwObARmWGu/rY1Q1to51tpxPp+vNt5ORALc4KRoZo8fwgujenG4uJQ7\npi7jp68uZX22fok8E90JLCIBpbisnGnf7OTFr9I4UFjKNT3b8B+XdyKuWYTT0eqNX28FYYwZAYxI\nSkq6Jy0tzek4IuKHDh0tZfLcrUxdtI3yCsttA9vxwCVJNIsMczpanfPrAjhOIwARqak9+Uf56xdb\neHdFJpFhIdx3UQfuGpxIw7DAXTGkAhAROcmWvQX88bNN/HtjDk0jQrkpJY7R/eNJjI50OlqtUwGI\niPyI5dv3M3XhNv61YS/lFZYhSdGMGRDPpV1aBsyziv26ADQHICJ1be+ho8xYnsnby3ayK/8oLaIa\nMKpfHKP6x9PGz7eY8OsCOE4jABGpa2XlFczdnMu0pTuYuyUXA1xyXkvGDIxnaHIMwUH+9+D6+tgL\nSETE74UEB3Fpl5Zc2qUlmfsLeXvZTmakZvLvjXuJbdqQ0f3juTkljpioBk5HrXUaAYiInKKkrILP\nv93DtKU7+CZjP6HBhsu7tuKnAxMYkNgMY9w9KtBHQCIitSA95zBvLd3JzBWZHDpaRoeYSMYMSOCG\nPrGu3YXUrwtAk8Ai4jZFJeV8tHYX05buZHXmQcJDgxjRow1jBibQM9bnqlGBXxfAcRoBiIgbrc/O\nZ9rSnXy4OpvCknK6tmnMTwcmcE3PNkQ2cH5qVQUgIlLHCo6WMmv1LqZ9s4NNewpo1CCE63q3ZczA\neM5r1dixXCoAEZF6Yq1l5c4DTPtmJx+t201JWQUpCU0ZMzCeK7q1rvcH1agAREQcsP9ICe+tyGLa\n0h1szyt0ZNsJvy4ATQKLiL+rqLAs3prHtKU7vrftxE8HxjOsc91uO+HXBXCcRgAiEgjqe9sJFYCI\niMucaduJC5NjCKqlbSe0FYSIiMucbduJWwfEc1Pf+tt2QiMAEREHnW7biV8PP++cH2OpEYCIiB8I\nCwliRM82jOjZ5sS2E3PW7iKiHp5YphGAiIjLlJVXEFKDVUJVHQG48vE3xpgRxpgp+fn5TkcREal3\nNfnPvzpcWQDW2jnW2nE+n8/pKCIiAcuVBSAiInVPBSAi4lEqABERj1IBiIh4lApARMSjVAAiIh7l\nyjuBj28HDRwyxpy8H7QP+LGbA37seDSwr24SVtvpctf3+1Xn+6py7tnOOdOf++O1dMt1rO731uW1\n9MfrCIF/LROq9G7WWr/5AqZU9TiQ6nTes+Wu7/erzvdV5dyznXOmP/fHa+mW6+ima+mP19Hr1/Lk\nL3/7CGhONY+7RW3nO9f3q873VeXcs51zpj/3x2vplutY3e+ty2vpj9cRvH0tT3D1XkA1YYxJtVXY\nC0PcT9cyMOg6uo+/jQCqY4rTAaTW6FoGBl1HlwnYEYCIiJxZII8ARETkDFQAIiIepQIQEfEozxSA\nMSbSGPO6MebvxpgxTueRc2OMaW+MedUYM9PpLFIzxphrK/89vmOM+YnTebzIrwvAGDPVGJNjjFl/\nyvHhxpjNxph0Y8zjlYevB2Zaa+8Brqn3sHJa1bmO1toMa+1YZ5LK2VTzWs6q/Pd4H3CLE3m9zq8L\nAHgNGH7yAWNMMDARuALoAow2xnQBYoHMytPK6zGjnN1rVP06iru9RvWv5ZOVfy71zK8LwFo7H9h/\nyuH+QHrlb4olwHRgJJDFsRIAP/97B5pqXkdxsepcS3PMc8Cn1tqV9Z1VAvM/wrZ895s+HPuPvy3w\nPnCDMeZl3H+bupzmOhpjmhtjJgO9jTFPOBNNqul0/yYfBC4FbjTG3OdEMK9z5W6gdcFaewS40+kc\nUjPW2jyOfWYsfs5aOwGY4HQOLwvEEUA2EHfS69jKY+JfdB0Dh66lSwViASwHko0xicaYMGAUMNvh\nTFJ9uo6BQ9fSpfy6AIwxbwNLgE7GmCxjzFhrbRnwAPA5sBGYYa391smccma6joFD19K/aDM4ERGP\n8usRgIiInDsVgIiIR6kAREQ8SgUgIuJRKgAREY9SAYiIeJQKQETEo1QAIiIepQIQEfGo/wf090WK\n/kNqZAAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x1155f6a58>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Figure 1 uses the right edges of bins where the width is more than one\n",
    "x = np.hstack((bins[:5], bins[6:]-1))\n",
    "plt.loglog(x, binned_sfs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[ 0.35831355  0.0691217   0.02810674  0.01868535  0.0142401   0.01123913\n",
      "  0.00419944  0.00194824  0.001405    0.00074064  0.00042244]\n"
     ]
    }
   ],
   "source": [
    "print(binned_sfs)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Here's what the SFS looks like in the paper:\n",
    "![Figure 2](img/gazave_fig2.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We are correctly reproducing the shape of the SFS, including the \"bumps\" at 10 and 100. However, we are showing too few singletons: 35.8% of variants vs. almost 40%."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Conclusions\n",
    "\n",
    "Overall, the data looks roughly as reported in the paper. Importantly, all SNVs are within target regions and we were able to reproduce the shape of the SFS. However there are a few inconsistencies:\n",
    "1. For some targets, the number of bases with called genotypes is larger than the target length. This is presumably due to some sort of overhang, where more bases were sequenced than targeted. However, this excess is never more than 4 bases and no SNVs were reprorted outside of the target regions. We will need to think about this a bit when making assumptions about masked vs. monomorphic sites.\n",
    "2. The maximum and average lengths of groups of nearby targets are slightly (~1%) different from what's reported in the SI. It would be nice to understand this, but I don't think it factors into our analysis at all.\n",
    "3. The fraction of SNVs with at least 450 called genotypes is higher than reported (99.8% vs. 95%). This may be because the SNVs in this file have not undergone the individual genotype filtering as reported on SI p. 9.\n",
    "4. The fraction of singletons sites is slightly lower than reported (37.8% vs. 38.4%). The discrepancy appears larger after downsampling (35.8%). This could be due to the differences in individual genotype filtering as in 3."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
